File: src/LiteGraph.Server/Classes/ChatModelSummary.cs
namespace LiteGraph.Server.Classes
{
    using System;

    /// <summary>
    /// Non-privileged projection of a chat endpoint, exposing only what a chat
    /// user needs to pick a model: identity, display name, model, provider,
    /// type, and whether it is the tenant default.  Endpoint URLs, keys, and
    /// health configuration are never included.
    /// </summary>
    public class ChatModelSummary
    {
        #region Public-Members

        /// <summary>
        /// Endpoint GUID, supplied as CompletionEndpointGUID or EmbeddingEndpointGUID on completion requests.
        /// </summary>
        public Guid GUID { get; set; } = Guid.Empty;

        /// <summary>
        /// Human-readable endpoint name.
        /// </summary>
        public string Name { get; set; } = null;

        /// <summary>
        /// Model identifier used by the provider.
        /// </summary>
        public string Model { get; set; } = null;

        /// <summary>
        /// Provider type.
        /// </summary>
        public ChatProviderTypeEnum Provider { get; set; } = ChatProviderTypeEnum.OpenAI;

        /// <summary>
        /// Endpoint type (Completion or Embedding).
        /// </summary>
        public ChatEndpointTypeEnum EndpointType { get; set; } = ChatEndpointTypeEnum.Completion;

        /// <summary>
        /// True when this endpoint is the tenant default for its type.
        /// </summary>
        public bool IsDefault { get; set; } = false;

        #endregion

        #region Private-Members

        #endregion

        #region Constructors-and-Factories

        /// <summary>
        /// Instantiate.
        /// </summary>
        public ChatModelSummary()
        {
        }

        /// <summary>
        /// Project a chat endpoint into a model summary.
        /// </summary>
        /// <param name="endpoint">Chat endpoint.  Must not be null.</param>
        /// <param name="isDefault">True when the endpoint is the tenant default for its type.</param>
        /// <returns>Model summary.</returns>
        /// <exception cref="ArgumentNullException">Thrown when endpoint is null.</exception>
        public static ChatModelSummary FromEndpoint(ChatEndpoint endpoint, bool isDefault)
        {
            if (endpoint == null) throw new ArgumentNullException(nameof(endpoint));

            return new ChatModelSummary
            {
                GUID = endpoint.GUID,
                Name = endpoint.Name,
                Model = endpoint.Model,
                Provider = endpoint.Provider,
                EndpointType = endpoint.EndpointType,
                IsDefault = isDefault
            };
        }

        #endregion

        #region Public-Methods

        #endregion

        #region Private-Methods

        #endregion
    }
}

File: sdk/js/src/models/ChatModelSummary.js
/**
 * ChatModelSummary class representing a non-privileged projection of a chat endpoint,
 * exposing only what a chat user needs to pick a model. Endpoint URLs, keys, and
 * health configuration are never included.
 */
export default class ChatModelSummary {
  /**
   * @param {Object} summary - Information about the model summary.
   * @param {string} [summary.GUID] - Endpoint GUID, supplied as CompletionEndpointGUID or EmbeddingEndpointGUID on completion requests.
   * @param {string|null} [summary.Name=null] - Human-readable endpoint name (default is null).
   * @param {string|null} [summary.Model=null] - Model identifier used by the provider (default is null).
   * @param {string} [summary.Provider='OpenAI'] - Provider type: OpenAI, Ollama, Gemini, Anthropic, or VoyageAI (default is OpenAI).
   * @param {string} [summary.EndpointType='Completion'] - Endpoint type: Embedding or Completion (default is Completion).
   * @param {boolean} [summary.IsDefault=false] - Indicates whether this endpoint is the tenant default for its type (default is false).
   */
  constructor(summary = {}) {
    const {
      GUID = null,
      Name = null,
      Model = null,
      Provider = 'OpenAI',
      EndpointType = 'Completion',
      IsDefault = false,
    } = summary;

    this.GUID = GUID; // Unique identifier for the chat endpoint
    this.Name = Name; // Human-readable endpoint name
    this.Model = Model; // Model identifier used by the provider
    this.Provider = Provider; // Provider type
    this.EndpointType = EndpointType; // Endpoint type (Embedding or Completion)
    this.IsDefault = IsDefault; // Indicates if this endpoint is the tenant default for its type
  }
}
